File: solutions/cpp/easy/c-tutorial-basic-data-types/solution-approach1.cpp
// ──────────────────────────────────────────────────
// Link        https://www.hackerrank.com/challenges/c-tutorial-basic-data-types/problem?isFullScreen=true
// Problem     Basic Data Types
// Difficulty  Easy
// Subdomain   Introduction
// Platform    HackerRank
// Language    cpp
// Status      Accepted
// Submitted   2026-08-20, 04:03 p.m.
// ──────────────────────────────────────────────────

#include <iostream>
#include <cstdio>
using namespace std;

int main() {
    // Complete the code.
    int a;
    float b;
    char c;
    long int d;
    double e;
    scanf("%d %ld %c %f %lf", &a,&d,&c,&b,&e);
    printf("%d \n%ld \n%c \n%f \n%lf", a,d,c,b,e);
    return 0;
} 

File: solutions/python3/easy/find-second-maximum-number-in-a-list/solution-approach1.py
# ──────────────────────────────────────────────────
# Link        https://www.hackerrank.com/challenges/find-second-maximum-number-in-a-list/problem?isFullScreen=true
# Problem     Find the Runner-Up Score!  
# Difficulty  Easy
# Subdomain   Basic Data Types
# Platform    HackerRank
# Language    python3
# Status      Accepted
# Submitted   2026-09-23, 07:29 a.m.
# ──────────────────────────────────────────────────

if __name__ == '__main__':
    n = int(input())
    arr = list(map(int, input().split()))
    fg=arr[0]
    sg=0
    for i in arr:
        
        if i>fg :
            sg=fg
            fg=i
        elif i>sg and i!=fg:
            sg=i
        elif sg==0 and i<0:
            sg=min(sg,i)
    print(sg)
